src/main/java/org/folio/circulation/infrastructure/storage/inventory/ItemRepository.java

```diff
@@ -254,6 +257,20 @@ private CompletableFuture<Result<MultipleRecords<Item>>> fetchItems(Collection<S
     final var mapper = new ItemMapper();
 
     return finder.findByIds(itemIds)
+      .thenApply(mapResult(identityMap::add))
+      .thenApply(mapResult(records -> records.mapRecords(mapper::toDomain)))
+      .thenCompose(res -> res.value().getTotalRecords() == itemIds.size()
+        ? CompletableFuture.completedFuture(res)
+        : res.combineAfter(x -> lookupDcbItem(res, itemIds), MultipleRecords::combine));
+  }
+
+  private CompletableFuture<Result<MultipleRecords<Item>>> lookupDcbItem(Result<MultipleRecords<Item>> inventoryItems, Collection<String> itemIds) {
+    var inventoryItemIds = inventoryItems.value().toKeys(Item::getItemId);
+    final var finder = new CqlIndexValuesFinder<>(createCirculationItemFinder());
+    var dcbItemIds = itemIds.stream().filter(ids -> !inventoryItemIds.contains(ids)).toList();
+    final var mapper = new ItemMapper();
+
+    return finder.findByIds(dcbItemIds)
```

Comment:
Consider this scenario: someone passed a list of 10 item IDs as a parameter to `fetchItems(Collection<String> itemIds)`, but they're not unique (two of them are the same). As a result, `dcbItemIds` here will be empty. Do we need to treat this case separately or does `finder.findByIds()` handle such cases by itself? Should we call `lookupDcbItem()` at all in such case?

Reply:
I think findByIds handle this case that is if itemIds are empty then it will return empty result. 
**if (criteria.getValues().isEmpty()) {
      return completedFuture(of(MultipleRecords::empty));
    }**